doc/manual/source/man/dnst-keyset.rst

```diff
@@ -0,0 +1,426 @@
+
+HSM Support (KMIP)
+~~~~~~~~~~~~~~~~~~
+
+Importing Keys
+~~~~~~~~~~~~~~
+
+There are three basic ways to import exiting keys: public-key,
+a public/private key pair from files or a public/private key pair in an HSM.
+
+A public key can only be import from a file.
+When the key is imported the name of the file is stored in the key set and
+the key will be included in DNSKEY RRset.
+This is useful for certain migration and to manually implement a
+multi-signer DNSSEC signing setup.
+Note that automation does not work for the case.
+
+A public/private key pair can be imported from files.
+It sufficient to give the name of the file that holds the public key if
+the filename ends in ``.key`` and the filename is the private key is the
+same except that it ends in ``.private``.
+If this is not the case then the private key filename can be specified
+separately.
+
+Importing a public/private key on an HSM require specifying the KMIP
+server ID, the ID of the public key, the ID of the private key, the
+DNSSEC algorithm of the key and the flags (typically 256 for a ZSK and
+257 for a KSK).
+
+Normally, keyset assumes ownership of any keys it hold.
+This mean that when a key is deleted from the key set, the keyset subcommand
+will also delete the files that hold the public and private or delete the
+key from the HSM.
+
+For an import public/private key pair this is considered too dangerous
+because another signer may need the keys.
+For this reason keys are imported in so-called ``decoupled`` state.
+When a decoupled key is deleted, only the reference to the key is deleted
+from the key set, the underlying keys are left untouched.
+There is a ``--coupled`` option to tell keyset to take ownership of the key.
+
+
+Migration
+~~~~~~~~~
+
+The keyset subcommand has no direct support for migration.
+Migration has to be done manually using the import commands.
+The semeantics of the import commands is decribed in the previous section.
+This section focusses on how the import command can be used to perform a
+migration.
+
+There are three migration strategies: 1) importing the existing signer's
+(private) signing keys, 2) a full multi-signer migration and 3)
+a partial multi-signer migration.
+
+Importing the existing signer's signing keys
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Importing the existing signer's public/private keys pairs is the easiest
+migration mechanism.
+The basic process is the following:
+
+* Disable (automatic) key rolls on the existing signer
+
+* Disable automatic key rolls before the executing the create command.
+
+* Import the KSK and ZSK (or CSK) as files or using KMIP between the
+  create and init commands.
+
+* Check with tools such as ldns-verify-zone that the new zone is secure with
+  the existing DS record at the parent.
+
+* Switch the nameservers to the new signer.
+
+* Perform key rolls for the KSK and ZSK (or the CSK).
+
+* (If needed) enable automatic key rolls.
```

Comment:
Should that be wanted rather than needed?

Reply:
Yes